shaders/src/common_defines.glsl

```diff
@@ -22,3 +22,13 @@
 
 #define float3x3 mat3
 #define float4x4 mat4
+
+// To workaround an adreno crash (#5294), we need ensure that a method with
+// parameter 'const mat4' does not call another method also with a 'const mat4'
+// parameter (i.e. mulMat4x4Float3). So we remove the const modifier for
+// materials compiled for vulkan+mobile.
+#if defined(TARGET_VULKAN_ENVIRONMENT) && defined(TARGET_MOBILE)
+   #define highp_mat4 highp mat4
+#else
+   #define highp_mat4 const highp mat4
+#endif
```

Comment:
what about we do this for everything everywhere and all types?

also, do you think we could do this at the SPIRV level? i.e. could we patch the SPIRV only on Adreno?

I think we should have a small spirv util lib we can use to manipulate spirv, I think it's going to become needed anyways.

Reply:
Do you mean take out all of the const modifier to the parameters? 

So interestingly, taking out the const in this case does seem to result in one extra store and one load in the resulting spirv. Though that might get optimized away through the optimizer. But I still hesitate to make this a larger change than necessary.